diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/KotlinParser.kt

```diff
@@ -0,0 +1,79 @@
+import org.jetbrains.kotlin.com.intellij.openapi.util.UserDataHolderBase
+import org.jetbrains.kotlin.com.intellij.pom.PomModel
+import org.jetbrains.kotlin.com.intellij.pom.PomModelAspect
+import org.jetbrains.kotlin.com.intellij.pom.PomTransaction
+import org.jetbrains.kotlin.com.intellij.pom.impl.PomTransactionBase
+import org.jetbrains.kotlin.com.intellij.pom.tree.TreeAspect
+import org.jetbrains.kotlin.com.intellij.psi.TokenType.ERROR_ELEMENT
+import org.jetbrains.kotlin.config.CompilerConfiguration
+import org.jetbrains.kotlin.psi.KtPsiFactory
+import sun.reflect.ReflectionFactory
+
+class KotlinParser {
+
+    /**
+     * Set idea.io.use.nio2 in system property to true
+     * fixme Maybe should check OS to don't change system property
+     */
+    init {
+        setIdeaIoUseFallback()
+    }
+
+    companion object{
+        private const val ONE_CHILD = 1
+    }
+
+    fun createNode(text: String, isPackage: Boolean = false): ASTNode {
+        val compilerConfiguration = CompilerConfiguration()
+        compilerConfiguration.put(CLIConfigurationKeys.MESSAGE_COLLECTOR_KEY, MessageCollector.NONE) // mute the output logging to process it themselves
+        val pomModel: PomModel = object : UserDataHolderBase(), PomModel {
+            override fun runTransaction(transaction: PomTransaction) {
+                (transaction as PomTransactionBase).run()
+            }
+            @Suppress("UNCHECKED_CAST")
+            override fun <T : PomModelAspect> getModelAspect(aspect: Class<T>): T? {
+                if (aspect == TreeAspect::class.java) {
+                    val constructor = ReflectionFactory.getReflectionFactory().newConstructorForSerialization(
+                            aspect,
+                            Any::class.java.getDeclaredConstructor(*arrayOfNulls<Class<*>>(0))
+                    )
+                    return constructor.newInstance() as T
+                }
+                return null
+            }
+        } // I don't really understand what's going on here, but thanks to this, you can use this node in the future
+        val project = KotlinCoreEnvironment.createForProduction(
+                Disposable {},
+                compilerConfiguration,
+                EnvironmentConfigFiles.JVM_CONFIG_FILES
+        ).project // create project
+        project as MockProject
+        project.registerService(PomModel::class.java, pomModel)
+        if (text.isEmpty())
+            throw KotlinParseException("Your text is not valid")
+        val ktPsiFactory = KtPsiFactory(project, true)
+        if (text.trim().isEmpty())
+            return ktPsiFactory.createWhiteSpace(text).node
+        lateinit var node: ASTNode
+        node = if (isPackage)
+            ktPsiFactory.createFile(text).node
+        else
+            ktPsiFactory.createBlockCodeFragment(text, null).node.findChildByType(BLOCK)!!
+        if (node.getChildren(null).size == ONE_CHILD)
```

Comment:
Why this strange constant? It is allowed to use `1`, it is not considered as a magic number.